src/MongoDB.Driver/Linq/FieldExpressionFlattener.cs

```diff
@@ -86,5 +79,16 @@ protected internal override Expression VisitField(FieldExpression node)
 
             return node;
         }
+
+        // private methods
+        private static string FormatArrayIndex(object index)
+        {
+            // We've treated -1 as meaning $ operator. We can't break this now,
+            // so, specifically when we are flattening fields names, this is 
+            // how we'll continue to treat -1.
+            return index is int convertedIndex && convertedIndex == -1
```

Comment:
I like this approach of no longer calling ToString at all.

Out of curiosity... I wonder if we need to handle `long` also? Not sure if an array index even CAN be a long though.

Reply:
I would suggest renaming `convertedIndex` to `intIndex`. Specially if we end up needing to handle long because then we'll probably need a `longIndex` also.

Reply:
I don't know about possible restrictions for this operation from the SERVER, but I would expect that long values also can be used as `positional` operator. So, I agree with it and will do it in next commit.

> a longIndex **also**.
do you think that we will need to have `int` and `long` validations together? I guess having only `longIndex` will be enough?

Reply:
But maybe I'm wrong, I need to check it